packages/yoastseo/src/languageProcessing/researches/readingTime.js

```diff
@@ -36,24 +40,38 @@ export default function( paper ) {
 		sv: 199,
 		tr: 166,
 	};
-	const minutesPerImage = 0.2;
 
-	let wordsPerMinuteScore = wordsPerMinute[ language ];
+	const charactersPerMinute = {
+		ja: 357,
+	};
+
 
-	// If the language is not on the list, assign the average of all language-dependent reading times as the score.
-	if ( ! wordsPerMinuteScore ) {
+	const wordsPerMinuteScore = wordsPerMinute[ language ];
+	const charactersPerMinuteScore = charactersPerMinute[ language ];
+
+	let wordsOrCharsPerMinuteScore;
+
+	if ( isFeatureEnabled( "JAPANESE_SUPPORT" ) && charactersPerMinuteScore ) {
+		wordsOrCharsPerMinuteScore = charactersPerMinuteScore;
+	} else if ( wordsPerMinuteScore ) {
+		wordsOrCharsPerMinuteScore = wordsPerMinuteScore;
+	} else {
+		// If the language is not on both lists, assign the average of all language-dependent reading times as the score.
 		const sumWordsPerMinute = Object.values( wordsPerMinute ).reduce( ( a, b ) => a + b );
 		const sumNumberOfLanguages = Object.keys( wordsPerMinute ).length;
-		wordsPerMinuteScore = sumWordsPerMinute / sumNumberOfLanguages;
+		wordsOrCharsPerMinuteScore = sumWordsPerMinute / sumNumberOfLanguages;
 	}
 
-	const numberOfWords = wordCountInText( paper );
+	const minutesPerImage = 0.2;
 	const numberOfImages = imageCount( paper );
 
+	const textLength = countCharactersFromArray ? countCharactersFromArray( customGetWords( paper.getText() ) ) : wordCountInText( paper );
```

Comment:
If you calculate the minutesToReadText in the if-statement above, you can simplify this part as well, I think.